lib/kubernetes-deploy/runner.rb

```diff
@@ -97,7 +103,14 @@ def run
       predeploy_priority_resources(resources)
 
       phase_heading("Deploying all resources")
-      deploy_resources(resources, prune: true)
+      if PROTECTED_NAMESPACES.include?(@namespace)
```

Comment:
hmm yes this is a belt and suspenders situation, it feels a bit too defensive to me. If we end up in a situation where `@prune` here is anything but false there is a bug, if we want to be deffensive why not `raise` then. What I'm saying is this would silently avoid a bug, but not tell us. I rather we either trust the rest of the code and the tests or blow up entirely on a broken assumption.